### ISSUE TITLE
why 'x = x.mean(dim = 1)' ??

### ISSUE THREAD

**Issue by abeyang00** [lucidrains/vit-pytorch#83]
in vit class forward, 
 def forward(self, img):
        x = self.to_patch_embedding(img)
        b, n, _ = x.shape

        cls_tokens = repeat(self.cls_token, '() n d -> b n d', b = b)
        x = torch.cat((cls_tokens, x), dim=1)
        x += self.pos_embedding[:, :(n + 1)]
        x = self.dropout(x)

        x = self.transformer(x)

        x = x.mean(dim = 1) if self.pool == 'mean' else x[:, 0]

        x = self.to_latent(x)
        return self.mlp_head(x)

why do we need to take average on first dimension??
what does that mean??

Thank you,
